### DISCUSSION TITLE
Improved and Detailed Documentation

### DISCUSSION THREAD

**Discussion by pratikmullick** [simeononsecurity/Windows-Optimize-Harden-Debloat#104] [Ideas]
The provided documentation is sparse, especially for two things:

1. Customization
2. Impact and Consideration

Both these pages refer back to the README, which provides generic HOWTOs.

For example, the associated links for issues Issue #22 refers to a [link](https://www.windowsphoneinfo.com/threads/windows-store-or-xbox-live-cant-sign-in-error-0x800706d9-enable-service-microsoft.273554/) that is suitable for XBox Live scenarios, while a better solution is provided [here](https://answers.microsoft.com/en-us/windows/forum/all/possible-fix-for-0x800706d9/c3fedcbc-0641-461a-b290-806dc4c4b5dc).

Can we discuss a framework or structure for improved documentation?

**Comment by simeononsecurity** [comment]
There are hundreds of thousands of lines of code in this project, it tracks and covers changes documented across many repos. 
It makes thousands of changes. As the readme reads, it's not really feasible to cover every edge case. 
We have automated testing to detect as many issues as possible. But if issues come up all we can do is point to documentation and provide guidance on any issues when they pop up.

This isn't intended for everyone. It was originally designed for personal use and others and I have spent time to make it more useable for others including at least 2 major rewrites. It isn't perfect. No hardening or debloating script is. 
You have to know what you're doing and expect that not everything is going to be documented. 

The error you're speaking of happens in multiple contexts. If you look at the related issues they each popped up in different scenarios ex the XBOX Live application and the Microsoft Store.  There were multiple ways to go about fixing it, it's best to check them all. 
If you need hand holding through everything, you're not the kind of person intended to run the script.

As far as documentation, if you want to submit a PR to improve any documentation, you're more than welcome to submit a pr. 
But given the workload, I'm not planning to document everything.  As issues pop up, I'll spend some time to look into things and document at a high level how to fix things where possible.
